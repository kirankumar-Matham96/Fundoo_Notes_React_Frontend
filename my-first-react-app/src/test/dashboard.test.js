/*********************************************************************
 * Execution    : cmd> npm start
 *
 * Purpose      : to test the dashboard components
 *
 * @description
 *
 * @file        : test/dashboard.test.js
 * @overview    : tests the different components whether they are rendered successfully or not
 * @module      : this is necessary to perform tests on the components for increasing performance of the code
 * @author      : Kirankumar Matham <mathamkirankumar96@gmail.com>
 * @version     : _ _ _
 * @since       : 22-06-2021
 *********************************************************************/

//importing all required libraries and components
import { shallow } from "enzyme";
import DashBoard from "../pages/dashboard";
import Header from "../components/dashboard.header";
import DropDown from "../components/dashboard.header.dropdownMenue";
import SideNav from "../components/dashboard.sideNav";

describe("Dashboard Testing", () => {
  test("Dashboard snapshot testing", () => {
    expect(shallow(<DashBoard />).length).toEqual(1);
  });

  test("Dashboard Header Test", () => {
    expect(shallow(<Header />).length).toEqual(1);
  });

  test("Dashboard Header Drop Down Menu Test", () => {
    expect(shallow(<DropDown />).length).toEqual(1);
  });

  test("Dashboard Side Nav Test", () => {
    expect(shallow(<SideNav />).length).toEqual(1);
  });
});
