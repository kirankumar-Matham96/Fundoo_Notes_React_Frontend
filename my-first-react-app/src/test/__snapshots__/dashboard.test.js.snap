// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing for Dashboard and It's Components givenDashboardComponentShallowWrapper_whenMatchWithSnapshotThatExists_shouldPassTheTest 1`] = `ShallowWrapper {}`;

exports[`Snapshot Testing for Dashboard and It's Components givenHeaderComponentShallowWrapper_whenMatchWithSnapshotThatExists_shouldPassTheTest 1`] = `ShallowWrapper {}`;

exports[`Snapshot Testing for Dashboard and It's Components givenNoteComponentShallowWrapper_whenMatchWithSnapshotThatExists_shouldPassTheTest 1`] = `ShallowWrapper {}`;

exports[`Snapshot Testing for Dashboard and It's Components givenSidenavComponentShallowWrapper_whenMatchWithSnapshotThatExists_shouldPassTheTest 1`] = `ShallowWrapper {}`;

exports[`Snapshot Testing for Register and Login pages givenDashboardComponentShallowWrapper_whenMatchWithSnapshotThatExists_shouldPassTheTest 1`] = `ShallowWrapper {}`;
